creepy/protocol/handshake.py

```diff
@@ -32,27 +32,36 @@ class HandshakeProtocol:
 
     def __init__(self, authorized_keys_path=None):
         self.salt = secrets.token_bytes(self.SALT_SIZE)
+        bobs = self._add_keys(authorized_keys_path)
+        self._bobs = bobs
+
+    def _add_key(self, key, bobs):
+        key_hash = self.pubkey_digest(key, self.salt)
+        bobs[key_hash] = Bob(key)
+
+    def _add_keys(self, authorized_keys_path):
         if authorized_keys_path is None:
             authorized_keys_path = '~/.ssh/authorized_keys'
         authorized_keys_path = os.path.expanduser(authorized_keys_path)
         bobs = {}
+        try:
+            self._add_keys_from_file(authorized_keys_path, bobs)
+        except IOError:
+            warnings.warn(f"Unable to read authorized keys file at '{authorized_keys_path}'")
+        ssh_dir = os.path.dirname(authorized_keys_path)
+        try:
+            self._add_key(load_public_key(ssh_dir=ssh_dir), bobs)
+        except Exception:
+            pass
+        return bobs
 
-        def add_key(key):
-            key_hash = self.pubkey_digest(key, self.salt)
-            bobs[key_hash] = Bob(key)
-
+    def _add_keys_from_file(self, authorized_keys_path, bobs):
```

Comment:
I seems better to make function
```python
def _load_public_keys(path: str) -> Iterator[PublicKeyType?]:
    ...
    yield load_public_key(line.strip())
    ....
```
that enumerates all keys from a file.

Reply:
If I do so, I will have to add another cycle like this:
```
for key in _load_public_keys(path):
    self._add_bob(key)
```
Isn't it worse? Or do I miss the idea?